subsys/random/rand32_timer.c

```diff
@@ -47,6 +47,16 @@ uint32_t z_impl_sys_rand32_get(void)
 	return val;
 }
 
+uint8_t z_impl_sys_rand8_get(void)
+{
+	return (uint8_t)z_impl_sys_rand32_get();
```

Comment:
Shouldn't you call `z_impl_sys_rand8_get()` here?
And similar comment to rand16 below.

Reply:
for the timer we use the implementation for 32 bit as it wouldn't be that much more efficient to have a custom implementation for 8 bit as it also would have to use `k_cycle_get_32()`. The aim of this PR is to optimise the use of the entropy device. The other entropy sources also get a implementation for 8 and 16 bit, but these will use the existing implementation of `sys_rand32_get()`, to be compatible.